avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroSerDeConfig.java

```diff
@@ -46,15 +46,24 @@
   public static final String AUTO_REGISTER_SCHEMAS_DOC =
       "Specify if the Serializer should attempt to register the Schema with Schema Registry";
 
+  public static final String BASIC_AUTH_CREDENTIALS_SOURCE = "basic.auth.credentials.source";
+  public static final String BASIC_AUTH_CREDENTIALS_SOURCE_DEFAULT = "URL";
+  public static final String BASIC_AUTH_CREDENTIALS_SOURCE_DOC =
+      "Specify how to pick the credentials for Basic uth header. "
+      + "The supported values are URL and SASL_INHERIT";
+
+  public static final String SCHEMA_REGISTRY_USER_INFO = "schema.registry.url.user.info";
+
   public static ConfigDef baseConfigDef() {
     return new ConfigDef()
         .define(SCHEMA_REGISTRY_URL_CONFIG, Type.LIST,
                 Importance.HIGH, SCHEMA_REGISTRY_URL_DOC)
         .define(MAX_SCHEMAS_PER_SUBJECT_CONFIG, Type.INT, MAX_SCHEMAS_PER_SUBJECT_DEFAULT,
                 Importance.LOW, MAX_SCHEMAS_PER_SUBJECT_DOC)
         .define(AUTO_REGISTER_SCHEMAS, Type.BOOLEAN, AUTO_REGISTER_SCHEMAS_DEFAULT,
-                Importance.MEDIUM, AUTO_REGISTER_SCHEMAS_DOC
-        );
+                Importance.MEDIUM, AUTO_REGISTER_SCHEMAS_DOC)
+        .define(BASIC_AUTH_CREDENTIALS_SOURCE, Type.STRING, BASIC_AUTH_CREDENTIALS_SOURCE_DEFAULT,
```

Comment:
Since we have a fixed set of options for this should we add a `ValidString` validator here?

Reply:
Also, should this be `Type.PASSWORD`? Presumably the main reason for having a separate config here is the password masking some users require?

Reply:
This is to specify how the basic auth credentials are provided. The default source would be URL but users could override it by specifying SASL_INHERIT

Reply:
oops, yeah, should have commented on the `SCHEMA_REGISTRY_USER_INFO` variable. regarding mutating, what I was thinking here was if `SCHEMA_REGISTRY_USER_INFO` is being filled in anyway, would it make sense to just have that as a `PASSWORD` config option? it gives users the opportunity to provide the config and have it masked *or* just include it in the URL where it wouldn't be masked. I'm thinking something along the lines of how the JDBC connector handles credentials settings: https://github.com/confluentinc/kafka-connect-jdbc/blob/master/src/main/java/io/confluent/connect/jdbc/source/JdbcSourceConnectorConfig.java#L48-L58 it can either be inline and unmasked, or as a separate config and masked.